trussworks/react-uswds
# feat: New Component ProcessList MVP

# Summary

This PR adds the [USWDS ProcessList component](https://designsystem.digital.gov/components/process-list/) in an MVP state.


## Related Issues or PRs

#979 

An enhancement that we may want to add as part of this GitHub issue would be to create a `ProcessListItemHeader` subcomponent. This would allow us to offer an out of the box way for our users to pass a heading that would automatically be styled by the `usa-process-list__heading` class irrespective of what element type the user passed. 

This enhancement would meet the USWDS Accessibility guideline, "Use semantic heading levels," that states, "Though our default code uses an `<h4>`, use the correct heading level with the class name `usa-process-list__heading` in your own implementation."


## How To Test

Check out this branch and run yarn test to execute tests and yarn storybook to see the new component in action.

Alternatively, you can scroll to the bottom of this page and click "Show environments" on the right above the comment input box to access this component in Storybook. 

### Screenshots (optional)

#### Default
![Screen Shot 2021-04-08 at 5 46 36 PM](https://user-images.githubusercontent.com/16230705/114112962-64e04a80-9892-11eb-9b41-79f8c97f4a30.png)

#### No text, custom sizing
![Screen Shot 2021-04-08 at 5 48 18 PM](https://user-images.githubusercontent.com/16230705/114113037-9ce78d80-9892-11eb-8799-21588d5b24b0.png)

#### Custom Sizing
![Screen Shot 2021-04-08 at 5 48 45 PM](https://user-images.githubusercontent.com/16230705/114113069-acff6d00-9892-11eb-9a12-ae39d4eecb47.png)
